packages/terra-signature/src/Signature.jsx

```diff
@@ -0,0 +1,198 @@
+  MEDIUM: 4,
+  HEAVY: 6,
+};
+const propTypes = {
+  /**
+   * The line width to use when drawing the signature on the canvas.
+   * One of LineWidths.EXTRA_FINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY.
+   */
+  lineWidth: PropTypes.oneOf([LineWidths.EXTRA_FINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY]),
+  /**
+  * Line segments that define signature.
+  */
+  lineSegments: PropTypes.array,
+  /**
+   * A callback function to execute when a line segment is drawn. The first parameter is the event, the
+   * second parameter is all the line segments, and the last parameter is the most recent line segment drawn.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  lineWidth: LineWidths.FINE,
+  lineSegments: [],
+  onChange: undefined,
+};
+
+class Signature extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = { lineSegments: props.lineSegments, lineWidth: props.lineWidth };
+
+    this.mouseDown = this.mouseDown.bind(this);
+    this.mouseUp = this.mouseUp.bind(this);
+    this.mouseXY = this.mouseXY.bind(this);
+    this.addLine = this.addLine.bind(this);
+    this.draw = this.draw.bind(this);
+    this.drawSignature = this.drawSignature.bind(this);
+    this.clearSignature = this.clearSignature.bind(this);
+    this.updateDimensions = this.updateDimensions.bind(this);
+  }
+
+  componentDidMount() {
+    if ('ontouchstart' in document.documentElement) {
+      this.canvas.addEventListener('touchstart', this.mouseDown, false);
+      this.canvas.addEventListener('touchmove', this.mouseXY, true);
+      this.canvas.addEventListener('touchend', this.mouseUp, false);
+      document.body.addEventListener('touchcancel', this.mouseUp, false);
+    } else {
+      this.canvas.addEventListener('mousedown', this.mouseDown);
+      this.canvas.addEventListener('mousemove', this.mouseXY);
+      document.body.addEventListener('mouseup', this.mouseUp);
+    }
+
+    const context = this.canvas.getContext('2d');
+
+    context.lineWidth = this.props.lineWidth;
+
+    const style = window.getComputedStyle(this.canvas);
+    const color = style.getPropertyValue('color');
+    context.strokeStyle = color;
+
+    this.updateDimensions();
+
+    window.addEventListener('resize', this.updateDimensions);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ((this.props.lineSegments !== nextProps.lineSegments) || (this.props.lineWidth !== nextProps.lineWidth)) {
+      this.setState({ lineSegments: nextProps.lineSegments, lineWidth: nextProps.lineWidth });
+      this.drawSignature(nextProps.lineSegments, nextProps.lineWidth);
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateDimensions);
+    window.removeEventListener('touchcancel', this.mouseUp);
+    window.removeEventListener('mouseup', this.mouseUp);
+  }
+
+  mouseDown(event) {
+    this.setState({ painting: true });
+
+    this.canvasRect = this.canvas.getBoundingClientRect();
```

Comment:
This was already set in updateDimensions

Reply:
When I fixed the problem with the Usage section in the README.md (to show examples of how to use the signature component), it had the side effect of pushing the signature canvas off screen.  And if the component mounts but the canvas is not visible, getBoundingClientRect() returns bogus values.  I need to re-evaluate getBoundingClientRect when the canvas becomes visible so I can correctly draw signature lines in the canvas.  It also needs to stay in updateDimensions() so that if the window resizes the canvas takes on its new width and height.